extensions/load-balancer/gce-extension/src/main/java/org/apache/stratos/gce/extension/util/GCEOperations.java

```diff
@@ -0,0 +1,588 @@
+
+        log.info("Adding instances to target pool: " + targetPoolName);
+
+        if (instancesIdsList.isEmpty()) {
+            log.warn("Cannot add instances to target pool. InstancesNamesList is empty.");
+            return;
+        }
+
+        List<InstanceReference> instanceReferenceList = new ArrayList<InstanceReference>();
+
+        //add instance to instance reference list, we should use the instance URL
+        for (String instanceId : instancesIdsList) { //for all instances
+
+            String instanceUrl = getInstanceURLFromId(instanceId);
+            if (instanceUrl != null) {
+                instanceReferenceList.add(new InstanceReference().
+                        setInstance(instanceUrl));
+            }
+
+        }
+
+        //create target pools add instance request and set instance to it
+        TargetPoolsAddInstanceRequest targetPoolsAddInstanceRequest = new
+                TargetPoolsAddInstanceRequest();
+        if (instanceReferenceList.isEmpty()) {
+            log.warn("Could not add instances to target pool " + targetPoolName + " because instance reference list is null");
+            return;
+        }
+
+        targetPoolsAddInstanceRequest.setInstances(instanceReferenceList);
+
+        try {
+            //execute
+            Operation operation = compute.targetPools().addInstance(PROJECT_ID, REGION_NAME,
+                    targetPoolName, targetPoolsAddInstanceRequest).execute();
+            waitForRegionOperationCompletion(operation.getName());
+
+        } catch (Exception e) {
+            log.warn("Could not add instance to target pool" + targetPoolName + " " + e);
+        }
+        log.info("Added instances to target pool: " + targetPoolName);
+    }
+
+    /**
+     * Create a health check in IaaS
+     *
+     * @param healthCheckName - name of the health check to be created
+     */
+    public void createHealthCheck(String healthCheckName) {
+
+        log.info("Creating health check: " + healthCheckName);
+
+        HttpHealthCheck httpHealthCheck = new HttpHealthCheck();
+        httpHealthCheck.setName(healthCheckName);
+        httpHealthCheck.setRequestPath(HEALTH_CHECK_REQUEST_PATH);
+        httpHealthCheck.setPort(Integer.parseInt(HEALTH_CHECK_PORT));
+        httpHealthCheck.setTimeoutSec(Integer.parseInt(HEALTH_CHECK_TIME_OUT_SEC));
+        httpHealthCheck.setCheckIntervalSec(Integer.parseInt(HEALTH_CHECK_INTERVAL_SEC));
+        httpHealthCheck.setUnhealthyThreshold(Integer.parseInt(HEALTH_CHECK_UNHEALTHY_THRESHOLD));
+        httpHealthCheck.setHealthyThreshold(Integer.parseInt(HEALTH_CHECK_HEALTHY_THRESHOLD));
+        try {
+            Operation operation = compute.httpHealthChecks().insert(PROJECT_ID, httpHealthCheck).execute();
+            waitForGlobalOperationCompletion(operation.getName());
+
+        } catch (Exception e) {
+            log.error("Could not create health check " + healthCheckName + " " + e);
```

Comment:
Please use the log.error(object,throwable) method when you are log the error logs. This is the standard we have followed through Stratos code base.